src/external/governance/GovernanceContract.sol

```diff
@@ -0,0 +1,339 @@
+    {
+        feeManager = newFeeManager;
+    }
+
+    //--------------------------------------------------------------------------
+    // Beacon Functions
+    //---------------------------
+    // Upgrade
+
+    /// @inheritdoc IGovernanceContract
+    function upgradeBeaconWithTimelock(
+        address beacon,
+        address newImplementation,
+        uint newMinorVersion
+    )
+        external
+        onlyCommunityOrTeamMultisig
+        accessableBeacon(beacon)
+        validAddress(newImplementation)
+    {
+        beaconTimelock[beacon] = Timelock(
+            true,
+            block.timestamp + timelockPeriod,
+            newImplementation,
+            newMinorVersion
+        );
+
+        emit BeaconTimelockStarted(
+            beacon,
+            newImplementation,
+            newMinorVersion,
+            block.timestamp + timelockPeriod
+        );
+    }
+
+    /// @inheritdoc IGovernanceContract
+    function triggerUpgradeBeaconWithTimelock(address beacon)
+        external
+        onlyCommunityOrTeamMultisig
+        accessableBeacon(beacon)
+        upgradeProcessAlreadyStarted(beacon)
+        timelockPeriodExceeded(beacon)
+    {
+        //set timelock to inaktive
+        beaconTimelock[beacon].timelockActive = false;
+
+        //Upgrade beacon
+        IInverterBeacon(beacon).upgradeTo(
+            beaconTimelock[beacon].intendedImplementation,
+            beaconTimelock[beacon].intendedMinorVersion,
+            false //this is not intended to override a shutdown
+        );
+
+        emit BeaconUpgraded(
+            beacon,
+            beaconTimelock[beacon].intendedImplementation,
+            beaconTimelock[beacon].intendedMinorVersion
+        );
+    }
+
+    /// @inheritdoc IGovernanceContract
+    function cancelUpgrade(address beacon)
+        external
+        onlyCommunityOrTeamMultisig
+        upgradeProcessAlreadyStarted(beacon)
+    {
+        beaconTimelock[beacon].timelockActive = false;
+        emit BeaconUpgradedCanceled(beacon);
+    }
+
+    /// @inheritdoc IGovernanceContract
+    function setTimelockPeriod(uint newTimelockPeriod)
+        external
+        onlyRole(COMMUNITY_MULTISIG_ROLE)
+        validAmount(newTimelockPeriod)
+    {
+        timelockPeriod = newTimelockPeriod;
+        emit TimelockPeriodSet(newTimelockPeriod);
+    }
+
+    //---------------------------
+    //Emergency Shutdown
+
+    /// @inheritdoc IGovernanceContract
+    function initiateBeaconShutdown(address beacon)
+        external
+        onlyCommunityOrTeamMultisig
+        accessableBeacon(beacon)
+    {
+        IInverterBeacon(beacon).shutDownImplementation();
+        emit BeaconShutdownInitiated(beacon);
+    }
+
+    /// @inheritdoc IGovernanceContract
+    function forceUpgradeBeaconAndRestartImplementation(
+        address beacon,
+        address newImplementation,
+        uint newMinorVersion
+    )
+        external
+        onlyRole(COMMUNITY_MULTISIG_ROLE)
```

Comment:
The Interface states that both community and team multisig should be able to call this function. The comment probably needs to be corrected as this function (if I understand correctly) circumvents the timelock and so should only be callable by the community multisig.